lib/liquid/standardfilters.rb

```diff
@@ -10,7 +10,7 @@ module StandardFilters
       '"'.freeze => '&quot;'.freeze,
       "'".freeze => '&#39;'.freeze
     }
-    HTML_ESCAPE_ONCE_REGEXP = /["><']|&(?!([a-zA-Z]+|(#\d+));)/
+    HTML_ESCAPE_ONCE_REGEXP = /["><']|&(?!([a-zA-Z]+\d*|(#\d+)|(#[xX][\da-fA-F]+));)/
```

Comment:
Just because they both are written in Ruby doesn't mean they should have the same bugs... Also I use Jekyll, so I'd assume most Jekyll users don't care about what Rails does. Neither would Shopify customers. Seems like no one's even using these entities anyways, based on the fact the bug is still there in master.

Similarly I found a [bug](https://github.com/textmate/html.tmbundle/commit/9f812c89f4990a98391701caa77824c94860538f) with TextMate where it wasn't recognizing uppercase hex entities. I guess no one really tests their entities regexes fully :)

The reason I found this in the first place is because I'm [experimenting](https://github.com/kainjow/Jeqyll) with my own port of Liquid/Jekyll.